packages/@aws-cdk/aws-events-targets/lib/state-machine.ts

```diff
@@ -13,13 +13,29 @@ export interface SfnStateMachineProps {
    * @default the entire EventBridge event
    */
   readonly input?: events.RuleTargetInput;
+
+  /**
+   * The IAM role to be assumed to execute the State Machine
+   *
+   * @default - a new role will be created
+   */
+  readonly role?: iam.IRole;
 }
 
 /**
  * Use a StepFunctions state machine as a target for Amazon EventBridge rules.
  */
 export class SfnStateMachine implements events.IRuleTarget {
+  private readonly role: iam.IRole;
```

Comment:
not sure it needs to be an instance variable since it's only used in the constructor

Reply:
It is not just used in the constructor, we are using it in the `bind` method.